confluent_kafka/avro/__init__.py

```diff
@@ -36,11 +37,28 @@ def __init__(self, config, default_key_schema=None,
         elif schema_registry_url is not None:
             raise ValueError("Cannot pass schema_registry along with schema.registry.url config")
 
+        self._serializer = MessageSerializer(schema_registry,
+                                             self._load_name_strategy(config, "value.subject.name.strategy"),
+                                             self._load_name_strategy(config, "key.subject.name.strategy"))
+
         super(AvroProducer, self).__init__(config)
-        self._serializer = MessageSerializer(schema_registry)
         self._key_schema = default_key_schema
         self._value_schema = default_value_schema
 
+    @staticmethod
+    def _load_name_strategy(config, key):
```

Comment:
to be clear, is your recommendation to additional optional args to the ctor?  that's easy enough to get done; i only went this way because it matched the behavior of the class as it exists today.